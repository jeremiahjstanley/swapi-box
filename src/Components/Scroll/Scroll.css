
.scroll-title {
  font-style: italic;
}