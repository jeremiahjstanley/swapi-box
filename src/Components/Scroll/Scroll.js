import React from 'react';
import '../../reset.css'
import './Scroll.css'

const Scroll = ({scroll}) => {
  return (
    <div className='scroll'>
      <p className='scroll-text'>{scroll.scrollText}</p>
      <p className='scroll-title'>{scroll.title}</p>
      <p className='scroll-date'>{scroll.releaseDate}</p>
    </div>
  );
}

export default Scroll