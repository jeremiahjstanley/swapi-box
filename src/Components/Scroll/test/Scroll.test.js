import React from 'react';
import { shallow } from 'enzyme';
import Scroll from '../Scroll';

describe('Scroll Test', () => {

  it('should match the snapshot', () => {
    const mockScroll = {scrollText: 'crawl',
                        title: 'title',
                        releaseDate: 'year'};
    const wrapper = shallow(<Scroll scroll={mockScroll}/>);

    expect(wrapper).toMatchSnapshot()
  });
});