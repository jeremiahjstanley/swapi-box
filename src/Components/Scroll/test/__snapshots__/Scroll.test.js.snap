// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll Test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll
    scroll={
        Object {
            "releaseDate": "year",
            "scrollText": "crawl",
            "title": "title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="scroll-text"
>
          crawl
</p>,
        <p
          className="scroll-title"
>
          title
</p>,
        <p
          className="scroll-date"
>
          year
</p>,
      ],
      "className": "scroll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "crawl",
          "className": "scroll-text",
        },
        "ref": null,
        "rendered": "crawl",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "scroll-title",
        },
        "ref": null,
        "rendered": "title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "year",
          "className": "scroll-date",
        },
        "ref": null,
        "rendered": "year",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="scroll-text"
>
            crawl
</p>,
          <p
            className="scroll-title"
>
            title
</p>,
          <p
            className="scroll-date"
>
            year
</p>,
        ],
        "className": "scroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "crawl",
            "className": "scroll-text",
          },
          "ref": null,
          "rendered": "crawl",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "scroll-title",
          },
          "ref": null,
          "rendered": "title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "year",
            "className": "scroll-date",
          },
          "ref": null,
          "rendered": "year",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
